include/podio/detail/Pythonizations.h

```diff
@@ -0,0 +1,60 @@
+#ifndef PODIO_DETAIL_PYTHONIZATIONS_H
+#define PODIO_DETAIL_PYTHONIZATIONS_H
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <stdexcept>
+#include <string>
+
+namespace podio::detail::pythonizations {
+
+// Callback function for the subscript pythonization
+// Calls the `at` method and change exception type to IndexError if the index is out of range
+static inline PyObject* subscript(PyObject* self, PyObject* index) {
+  PyObject* result = PyObject_CallMethod(self, "at", "O", index);
+  if (!result) {
+    PyObject* exc = PyErr_Occurred();
+    // Check if the exception is `cppyy.gbl.std.out_of_range`
+    // Since PyImport_ImportModule("cppyy") fails, this workaround checks the exception name
+    if (exc && PyObject_HasAttrString(exc, "__name__")) {
+      PyObject* exc_name = PyObject_GetAttrString(exc, "__name__");
+      if (exc_name) {
+        const char* name_cstr = PyUnicode_AsUTF8(exc_name);
+        if (name_cstr && strcmp(name_cstr, "out_of_range") == 0) {
+          PyErr_Clear();
+          PyErr_SetString(PyExc_IndexError, "Index out of range");
+        }
+        Py_DECREF(exc_name);
+      }
+    }
+  }
+  return result;
+}
+
+// Helper to register the subscript pythonization callback as `__getitem__` method
+static inline void pythonize_subscript(PyObject* klass, const std::string& name) {
+  static PyMethodDef ml = {"subscipt_pythonization", subscript, METH_VARARGS, R"(
+        Raise an `IndexError` exception if an index is invalid.
+        The `__getitem__` will return immutable datatype objects instead of the mutable ones.
+  )"};
+  auto* func = PyCFunction_New(&ml, klass);
+  if (!func) {
+    throw std::runtime_error("Failed to create Python subscript function for class " + name);
```

Comment:
When will these throw if they throw? During `import XYZ`? Or earlier / later?

Reply:
Thanks for the clarification. I don't think we want to catch these as I wouldn't even be sure how we would recover from this. In the end we will more likely than not just leave the whole loaded cppyy module in a broken state in case we ever throw this exception(?).

But all the information that is spat out should at least point users towards an appropriate github repository to open an issue ;)